content/en/blog/_posts/2019-08-30-announcing-etcd-3.4.md

```diff
@@ -0,0 +1,64 @@
+
+![learner-figure-1](/images/blog/2019-08-30-announcing-etcd-3.4/figure-1.png)
+
+The worst case is the misconfigured membership add. etcd membership reconfiguration is a two-step process: `etcdctl member add` with peer URLs, and starting a new etcd to join the cluster. That is, `member add` command is applied whether the peer URL value is invalid or not. If the first step is to apply the invalid URLs and change the quorum size, it is possible that the cluster already loses the quorum until the new node connects. Since the node with invalid URLs will never become online and there’s no leader, it is impossible to revert the membership change (see *Figure 2*).
+
+![learner-figure-2](/images/blog/2019-08-30-announcing-etcd-3.4/figure-2.png)
+
+This becomes more complicated when there are partitioned nodes (see the [design document](https://github.com/etcd-io/etcd/blob/master/Documentation/learning/design-learner.md) for more).
+
+In order to address such failure modes, etcd introduces a [new node state “Learner”](https://github.com/etcd-io/etcd/issues/10537), which joins the cluster as a non-voting member until it catches up to leader’s logs. This means the learner still receives all updates from leader, while it does not count towards the quorum, which is used by the leader to evaluate peer activeness. The learner only serves as a standby node until promoted. This relaxed requirements for quorum provides the better availability during membership reconfiguration and operational safety (see *Figure 3*).
+
+![learner-figure-3](/images/blog/2019-08-30-announcing-etcd-3.4/figure-3.png)
+
+We will further improve learner robustness, and explore auto-promote mechanisms for easier and more reliable operation. Please read our [learner design documentation](https://github.com/etcd-io/etcd/blob/master/Documentation/learning/design-learner.md) and [runtime-configuration document](https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/runtime-configuration.md#add-a-new-member-as-learner) for user guides.
+
+## Better Storage Backend
+
+etcd v3.4 also includes a number of performance improvements for large scale Kubernetes workloads.
+
+Especially, etcd experienced performance issues with a large number of concurrent read transactions even when there is no write (e.g. `“read-only range request ... took too long to execute”`). Previously, storage backend commit operation on pending writes blocks incoming read transactions, even when there was no pending write. Now, the commit [does not block reads](https://github.com/etcd-io/etcd/pull/9296), to improve long-running read transaction performance.
```

Comment:
Especially -> In particular
Previously, storage -> Previously, the storage
read, to improve -> reads which improves